framework/16-inch/amd-ai-300-series/README.md

```diff
@@ -28,24 +44,66 @@ This enables hybrid graphics with PRIME offload: the AMD iGPU runs by default fo
 Find your bus IDs with:
 
 ```sh
-$ lspci | grep -E "VGA|3D|Display"
+$ nix-shell -p pciutils --run 'lspci | grep -E "VGA|3D|Display"'
 c2:00.0 VGA compatible controller: NVIDIA Corporation ...
 c3:00.0 Display controller: Advanced Micro Devices ...
 ```
 
 Convert the hex bus ID to decimal (e.g., `c2:00.0` → `PCI:194:0:0`, `c3:00.0` → `PCI:195:0:0`).
 
-See also [NVIDIA](https://wiki.nixos.org/wiki/NVIDIA) on the NixOS Wiki.
+Helper:
+
+```sh
+# Replace BDF with the lspci value like c2:00.0
+BDF="c2:00.0"
+BUS="${BDF%%:*}"; REST="${BDF#*:}"
+DEV="${REST%%.*}"; FUN="${REST#*.}"
+printf 'PCI:%d:%d:%d\n' "$((16#$BUS))" "$((16#$DEV))" "$FUN"
+```
+
+#### Validate hybrid / offload behavior
+
+```sh
+# Default should use AMD
+nix-shell -p mesa-demos --run 'glxinfo -B | grep -E "OpenGL vendor|OpenGL renderer"'
+
+# Offload should use NVIDIA
+nix-shell -p mesa-demos --run 'nvidia-offload glxinfo -B | grep -E "OpenGL vendor|OpenGL renderer"'
+```
+
+To check whether the NVIDIA GPU is runtime-suspending at idle:
 
-## Updating Firmware
+```sh
+# Replace with your NVIDIA PCI address (from `nix-shell -p pciutils --run 'lspci -D'`)
+cat /sys/bus/pci/devices/0000:??:??.?/power/runtime_status
```

Comment:
The needed variables are already set, so might also use `cat /sys/bus/pci/devices/0000:$BUS:$DEV.$FUN/power/runtime_status` instead?